### PR TITLE
Raspi

### PR DESCRIPTION
**Pull request by theilmbh**
This contains updates to allsummary.py to work with the new magpi system

Essentially, I:

- Added a call to an shell script that rsyncs all the magpis
- Changed a few paths to reflect magpi path structure
- works on python3

### CODE REVIEW

**Comment by MarvinT** [comment]
This will break it on zog because of the python3

**Comment by theilmbh** [comment]
Yes,  it absolutely will, that's why we keep this branch separate until zog doesn't exist.  

I should change this to rsync the pis individually based on panel_subject_behavior, and send emails if the rsync fails.

File: glab_common/allsummary.py [hunk 1]
```diff
@@ -1,11 +1,11 @@
-#!/usr/local/bin/python
+#!/usr/bin/python3
```
Comment (MarvinT) [inline glab_common/allsummary.py]:
```suggestion
```
I recommend removing the shebang alltogether and instead explicitly running the script with the python in the crontab.
The location of python on the computer isn't really part of the script.

File: glab_common/allsummary.py [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
 import re
 import datetime as dt
 from behav.loading import load_data_pandas
 import warnings
-from pyoperant.local import DATA_PATH
+import subprocess
 
-process_fname = DATA_PATH + "panel_subject_behavior"
+process_fname = "/home/bird/opdat/panel_subject_behavior"
```
Comment (MarvinT) [inline glab_common/allsummary.py]: I'm comfortable with this to remove the pyoperant dependency.

File: glab_common/allsummary.py [hunk 3]
```diff
@@ -24,6 +24,9 @@
             processes.append(spl_line[4])
 
 inf.close()
+
+# rsync magpis
+rsync_output = subprocess.run(["/home/bird/bin/rsync_magpis.sh"])
```
Comment (MarvinT) [inline glab_common/allsummary.py]: do we necessarily want the rsync to run from this script?
If we are, we should have the rsync script be part of this repo.

File: glab_common/allsummary.py [hunk 4]
```diff
@@ -96,11 +99,10 @@
                 minutes_ago = (dt.datetime.now() - last_trial_time).seconds / 60
                 datediff = '(%d mins ago)' % (minutes_ago)
             
-            outline = "box %d\tB%d\t %s  \ttrls=%s  \tfeeds=%d  \tTOs=%d  \tnoRs=%d  \tFeedErrs=(%s,%s,%s,%s)  \tlast @ %s %s\n" % (box, bird, proc, trials_run, feeder_ops, TOs, noRs, 
+            outline = "magpi %d\tB%d\t %s  \ttrls=%s  \tfeeds=%d  \tTOs=%d  \tnoRs=%d  \tFeedErrs=(%s,%s,%s,%s)  \tlast @ %s %s\n" % (box, bird, proc, trials_run, feeder_ops, TOs, noRs, 
               hopper_failures, godown_failures, goup_failures, resp_feed, last_trial_time.strftime('%x %X'), datediff)
             f.write(outline)
     except Exception as e:
         f.write("box %d\tB%d\t Error opening SummaryDat or incorrect format\n" % (box, bird))
-        print e
```
Comment (MarvinT) [inline glab_common/allsummary.py]: This is actually a really useful line that only matters when trying to debug. Especially since we catch all generic Exceptions. This can easily be updated to work in python 3.

File: glab_common/allsummary.py [hunk 5]
```diff
@@ -37,7 +40,7 @@
 for (box, bird, proc) in zip(box_nums, bird_nums, processes):
     try:
         if proc in ('shape','lights','pylights', 'lights.py'):
-            f.write("box %d\tB%d\t %s\n" % (box, bird, proc))
+            f.write("magpi %d\tB%d\t %s\n" % (box, bird, proc))
```
Comment (MarvinT) [inline glab_common/allsummary.py]: This seems like an unnecessary change. The boxes are still referring to boxes...

**Comment by theilmbh** [comment]
Let me make the changes described above then we can resolve conflicts.

**Comment by theilmbh** [comment]
@MarvinT can you take a look at latest commit?

**Comment by MarvinT** [comment]
looks good, shouldn't be too much of a conflict.
It might also be worth throwing a ```from __future__ import print_function``` at the top to suppress warnings in python 2

**Comment by MarvinT** [comment]
looks good to me. Does it run on magpi?
